packages/xo-server/src/index.js

```diff
@@ -381,18 +384,37 @@ async function makeWebServerListen(
     ...opts
   }
 ) {
-  if (cert && key) {
-    ;[opts.cert, opts.key] = await Promise.all([readFile(cert), readFile(key)])
-    if (opts.key.includes('ENCRYPTED')) {
-      opts.passphrase = await new Promise(resolve => {
-        // eslint-disable-next-line no-console
-        console.log('Encrypted key %s', key)
-        process.stdout.write(`Enter pass phrase: `)
-        pw(resolve)
-      })
-    }
-  }
   try {
+    if (cert && key) {
+      try {
+        ;[opts.cert, opts.key] = await Promise.all([
+          readFile(cert),
+          readFile(key),
+        ])
+        if (opts.key.includes('ENCRYPTED')) {
+          opts.passphrase = await new Promise(resolve => {
+            // eslint-disable-next-line no-console
+            console.log('Encrypted key %s', key)
+            process.stdout.write(`Enter pass phrase: `)
+            pw(resolve)
+          })
+        }
+      } catch (error) {
+        if (!(autoCert && error.code === 'ENOENT')) {
+          throw error
+        }
+
+        const pems = await genSelfSignedCert()
+        await Promise.all([
+          outputFile(cert, pems.cert, { flag: 'wx', mode: 0o400 }),
```

Comment:
If the file exists, it will throw an error which will override the initial error.

Reply:
Yes, because we know its a missing file (`ENOENT`) and this is a special handling for it.

Reply:
I understand now, so, i'll take back what i've said her https://github.com/vatesfr/xen-orchestra/pull/4954/files#r415794444 because we should always use the cert and the key provided by users and throw if they don't exists.